### ISSUE TITLE
More Pythonic force group specification in Context.getState

### ISSUE THREAD

**Issue by rmcgibbo** [pandegroup/openmm#1197]
The API for querying only particular force groups with `Context.getState` requires thinking about bit-wise operations, which is generally not familiar territory for Python programmers.

Current API:
```
Context.getState
[... snip ...]
- groups (int=-1) a set of bit flags for which force groups to include when computing
  forces and energies.  Group i will be included if (groups&(1<<i)) != 0.  The default
  value includes all groups.
```

It would be convenient to instead to be able to pass in a set (or maybe just an iterable) of integers for `groups`. Changing this in the C++ API requires dealing with backwards compatibility issues, but Python weak typing should make it possible to either accept an integer or a set for this argument.

I thought of this when responding to the following email / use case from @dr-nate:

> Yeah, there's the notion of "force groups" -- so what you want to do is put the particular force you're interested in in a separate force group (force.setForceGroup()), and then when you query the context with context.getState, ask for only that force group.
>
> -Robert
>
>> On Tue, Oct 20, 2015 at 2:28 PM, Nathaniel Stanley <> wrote:
>> Hey Robert,
>> 
>> Is there a way in OpenMM to print out the force an added external force is contributing? I know how to report the net force on an atom, but I was wondering about the contribution of a force I've applied. It would be a helpful way to check I'm doing it correctly.

>> Nate

**Comment by dr-nate** [comment]
Just to clarify on this for others that might find this before it's changed in the future, here's an example:

You set a group for a force with an integer between 0 and 31 (32 possible groups):
```
my_force.setForceGroup(1)
```

And then your force reporter would look like this:

```
class ForceReporter(object):
    def __init__(self, file, reportInterval):
        self._out = open(file, 'w')
        self._reportInterval = reportInterval

    def __del__(self):
        self._out.close()

    def describeNextReport(self, simulation):
        steps = self._reportInterval - simulation.currentStep%self._reportInterval
        # return (steps, positions, velocities, forces, energies)
        return (steps, False, False, True, False)

    def report(self, simulation, state):
        state = simulation.context.getState(getForces=True,groups=0b00000000000000000000000000000010)
        forces = state.getForces().value_in_unit(u.kilocalories_per_mole/u.angstroms)
        self._out.write(str(forces)+"\n")

force_reporter = ForceReporter('forces.txt', forcefreq)
simulation.reporters.append(force_reporter)
```
Note that the "groups" arguement takes a bitfield (series of zeroes or ones), where 1 means "compute the positions/energies/forces/etc. for this group, and 0 means don't. Each position in the bitfield represents a group. Also note that the bit field goes from right to left, so group 1 is in the second position from right.

**Comment by jchodera** [comment]
I am in favor of supporting more Pythonic idioms for everything, including
this.

**Comment by rmcgibbo** [comment]
I had no idea Python had binary literals like that.

**Comment by rmcgibbo** [comment]
Fixed by #1199
